### PR TITLE
Added documentation for monitoring

### PR DESCRIPTION
**Pull request by temitope777**
WHY: to enable the dev teams understand the process better
HOW: By updating the md file

## Context
Added the monitoring URLs and the procedure for creating dashboards

## Changes proposed in this pull request
added the descriptions to the md file
## Guidance to review
check md file that it meets requirements

### CODE REVIEW

**Review by VenkataChalla554 — APPROVED** [review]
Looks fine to me

File: documentation/monitoring.md [hunk 1]
```diff
@@ -75,6 +75,42 @@ There are several other variables that can be changed depending on env requireme
...
+### Creating a Grafana Dashboard from existing json
+
+- Log in to Grafana as admin
+- Navigate to the Dashboard Import Page and click the "+" icon in the left sidebar to open the dashboard menu, select "Import" from the  
+  dropdown menu to access the dashboard import page.
+- Import the JSON File by either clicking on "Upload JSON file" and selecting the json file from your computer or pasting the json file 
+  content into the text area provided
+-  Click on the "Import" button to initiate the dashboard import process
+
+
```
Comment (RMcVelia) [inline documentation/monitoring.md:113]: should explain that dashboard additions/updates will be overwritten and need to be PR'd to be permanent. List where/how you would add the dashboard via PR

File: documentation/monitoring.md [hunk 2]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
 - grafana_app_mem - app memory limit (default 1Gi)
 - grafana_app_cpu - app requests cpu (default 500m)
 
+Grafana could be accessed via the respective URL's based on the environment of interest. The URL's corresponding to each environment is as below
+
+### Test | https://grafana.test.teacherservices.cloud
```
Comment (saliceti) [inline documentation/monitoring.md:80]: Formatting:
- Test: https://grafana.test.teacherservices.cloud
- Production: https://grafana.teacherservices.cloud

File: documentation/monitoring.md [hunk 3]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
 - grafana_app_mem - app memory limit (default 1Gi)
 - grafana_app_cpu - app requests cpu (default 500m)
 
+Grafana could be accessed via the respective URL's based on the environment of interest. The URL's corresponding to each environment is as below
```
[the remark this answers is not in the document]
Comment (saliceti) [reply documentation/monitoring.md:78]: Grammar:
"are as below:"

File: documentation/monitoring.md [hunk 4]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
...
+
+### Test | https://grafana.test.teacherservices.cloud 
+### Production | https://grafana.teacherservices.cloud/
+
+The default access to the grafana interface is view only, which does not require authentication. In order to be able to make changes for example adding more dashboards and editing existing dashboards, requests will have to be made in the Teacher-services-Infra slack channel to obtain admin credentials
```
Comment (saliceti) [inline documentation/monitoring.md:83]: Teacher-services-Infra -> #teacher-services-infra

Comment (saliceti) [reply documentation/monitoring.md:83]: End sentence with a dot

File: documentation/monitoring.md (hunk 1 above)
Comment (temitope777) [reply documentation/monitoring.md:115]: done

File: documentation/monitoring.md [hunk 5]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
...
+-	Create Panels for each metric by adding panel for the metric and choosing the right visualisation (for example graph, gauge, table, 
+    heatmap) and configure the panel settings eg the query, data transformation and diaplay options and add concise title for clarity
+
+
+### Exporting a Grafana Dashboard as json
+Grafana allows you to export your dashboard as a JSON file, which can be version controlled and shared with others. This could be achieved by following these steps
+- 	Open your dashboard in Grafana
+-	Click on the "Share" button(icon) in the top left corner
+-	In the "Export" tab, select "Export for sharing externally"
+-	Click "Save to file" to download the JSON file of your dashboard
```
Comment (saliceti) [inline documentation/monitoring.md:102]: Add steps to PR the teacher-services-cloud repo and where to add the json file

File: documentation/monitoring.md [hunk 6]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
...
+
+### Creating a Grafana Dashboard from existing json and updating the code with new dashboard
+
+- Log in to Grafana as admin
+- Navigate to the Dashboard Import Page and click the "+" icon in the left sidebar to open the dashboard menu, select "Import" from the  
+  dropdown menu to access the dashboard import page.
+- Import the JSON File by either clicking on "Upload JSON file" and selecting the json file from your computer or pasting the json file 
+  content into the text area provided
+-  Click on the "Import" button to initiate the dashboard import process
+- in order to ensure that the new dashboard created is permanent and not deleted by subsequent deployment, create a JSON  file for the
```
Comment (saliceti) [inline documentation/monitoring.md:112]: Move to a new section

File: documentation/monitoring.md [hunk 7]
```diff
@@ -102,6 +140,14 @@ There are several other variables that can be changed depending on env requireme
 - alertmanager_app_mem - app memory limit (default 1G)
 - alertmanager_app_cpu - app cpu requests (default 1)
 
+The alertmanager urls corresponding to the various environments are
+### development | https://alertmanager.cluster[clusternumber].development.teacherservices.cloud/
```
Comment (saliceti) [inline documentation/monitoring.md:144]: development is not required here, as the developers don't need it

File: documentation/monitoring.md [hunk 8]
```diff
@@ -102,6 +140,14 @@ There are several other variables that can be changed depending on env requireme
...
+### Test | https://alertmanager.test.teacherservices.cloud/
```
Comment (saliceti) [inline documentation/monitoring.md:145]: Formatting:
- Test: https://alertmanager.test.teacherservices.cloud
- Production: https://alertmanager.teacherservices.cloud

File: documentation/monitoring.md [hunk 9]
```diff
@@ -102,6 +140,14 @@ There are several other variables that can be changed depending on env requireme
...
+### Production | https://alertmanager.teacherservices.cloud/
+where clusternumber is the number of the cluster from 1 to 6
+
+authentication details are usually required but this is stored in the keyvault. Please ask in the teacher-services-infra channel for more details
```
Comment (saliceti) [inline documentation/monitoring.md:149]: Formatting:
- Authentication
- End with a dot

File: documentation/monitoring.md [hunk 10]
```diff
@@ -75,6 +75,44 @@ There are several other variables that can be changed depending on env requireme
...
 
+Grafana could be accessed via the respective URL's based on the environment of interest. The URL's corresponding to each environment is as below
+
+### Test | https://grafana.test.teacherservices.cloud 
+### Production | https://grafana.teacherservices.cloud/
+
+The default access to the grafana interface is view only, which does not require authentication. In order to be able to make changes for example adding more dashboards and editing existing dashboards, requests will have to be made in the Teacher-services-Infra slack channel to obtain admin credentials 
+
+### Editing or creating new grafana dashboards
+The following steps are required for creating or editing dashboards
```
Comment (saliceti) [inline documentation/monitoring.md:86]: Can we link to the doc? https://grafana.com/docs/grafana/latest/fundamentals/dashboards-overview/

File: documentation/monitoring.md (hunk 7 above)
Comment (temitope777) [reply documentation/monitoring.md:144]: removed

File: documentation/monitoring.md (hunk 3 above)
Comment (saliceti) [inline documentation/monitoring.md:78]: This card is "Document monitoring URLs for devs"
This doc here is not for developers, it's infra team documentation. Help for devs should go to the [developer-onboarding.md](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/developer-onboarding.md) doc. Please move all the content there.

File: documentation/monitoring.md (hunk 9 above)
Comment (temitope777) [reply documentation/monitoring.md:149]: done

File: documentation/monitoring.md (hunk 3 above)
Comment (temitope777) [reply documentation/monitoring.md:78]: moved

File: documentation/monitoring.md (hunk 2 above)
Comment (temitope777) [reply documentation/monitoring.md]: formatted

File: documentation/monitoring.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (temitope777) [reply documentation/monitoring.md]: fixed

File: documentation/monitoring.md (hunk 4 above)
Comment (temitope777) [reply documentation/monitoring.md]: done

File: documentation/monitoring.md (hunk 6 above)
Comment (temitope777) [reply documentation/monitoring.md]: done

File: documentation/developer-onboarding.md [hunk 11]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
 ContainerLogV2
 | where LogEntry contains "cpd-production-cpd-ecf-production-web-80"
 ```
+## Monitoring
```
Comment (saliceti) [inline documentation/developer-onboarding.md:145]: Formatting: blank line above

File: documentation/developer-onboarding.md [hunk 12]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+
+The main monitoring tools used are Grafana and Alertmanager. For further reading about Monitoring setup in the cluster [click here](https://github.com/DFE-Digital/teacher-services-cloud/blob/97a19da1aedf604e7778b25ce747fd9a9a61a670/documentation/monitoring.md)
+### Grafana
```
Comment (saliceti) [inline documentation/developer-onboarding.md:148]: Formatting: blank line above

File: documentation/developer-onboarding.md [hunk 13]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+Grafana could be accessed via the respective URL's based on the environment of interest. The URLs corresponding to each environment are as below
```
Comment (saliceti) [inline documentation/developer-onboarding.md:149]: Typo: URLs

File: documentation/monitoring.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (saliceti) [reply documentation/monitoring.md]: there is another one

File: documentation/developer-onboarding.md [hunk 14]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
 ContainerLogV2
 | where LogEntry contains "cpd-production-cpd-ecf-production-web-80"
 ```
+## Monitoring
+
+The main monitoring tools used are Grafana and Alertmanager. For further reading about Monitoring setup in the cluster [click here](https://github.com/DFE-Digital/teacher-services-cloud/blob/97a19da1aedf604e7778b25ce747fd9a9a61a670/documentation/monitoring.md)
```
Comment (saliceti) [inline documentation/developer-onboarding.md:147]: End with a dot

File: documentation/developer-onboarding.md (hunk 13 above)
Comment (saliceti) [inline documentation/developer-onboarding.md:149]: as below:

File: documentation/developer-onboarding.md [hunk 15]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+### Grafana
+Grafana could be accessed via the respective URL's based on the environment of interest. The URLs corresponding to each environment are as below
+
+* Test | https://grafana.test.teacherservices.cloud 
+* Production | https://grafana.teacherservices.cloud/
+
+The default access to the grafana interface is view only, which does not require authentication. In order to be able to make changes for example adding more dashboards and editing existing dashboards, requests will have to be made in the #teacher-services-infra slack channel to obtain admin credentials
```
Comment (saliceti) [inline documentation/developer-onboarding.md:154]: End with a dot

File: documentation/developer-onboarding.md [hunk 16]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+-	In the "Export" tab, select "Export for sharing externally"
+-	Click "Save to file" to download the JSON file of your dashboard
+
+### Editing or creating new grafana dashboards
+The following steps are required for creating or editing dashboards. Please [click](https://grafana.com/docs/grafana/latest/fundamentals/dashboards-overview/) for more extensive details
+-   Ensure you are logged in as an admin
+-   Identify the purpose of your dashboard. What insights the dashboard will provide and what messages it  conveys
+- 	Plan and Design how the dashboard would look when completed, paying attention to the placement of panels, alignment, spacing, colour 
+    and organisation
+- 	Select the Appropriate Data Sources by identifying the right datasource to visualise in the dashboard (examples of usual datasources
```
Comment (saliceti) [inline documentation/developer-onboarding.md:169]: There is only 1 datasource here

File: documentation/developer-onboarding.md [hunk 17]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+
+### Editing or creating new grafana dashboards
+The following steps are required for creating or editing dashboards. Please [click](https://grafana.com/docs/grafana/latest/fundamentals/dashboards-overview/) for more extensive details
+-   Ensure you are logged in as an admin
+-   Identify the purpose of your dashboard. What insights the dashboard will provide and what messages it  conveys
+- 	Plan and Design how the dashboard would look when completed, paying attention to the placement of panels, alignment, spacing, colour 
+    and organisation
+- 	Select the Appropriate Data Sources by identifying the right datasource to visualise in the dashboard (examples of usual datasources 
+    are prometheus, , databases, APIs, log files)
+-	Create Panels for each metric by adding panel for the metric and choosing the right visualisation (for example graph, gauge, table,
```
Comment (saliceti) [inline documentation/developer-onboarding.md:171]: How can users find the available metrics?

File: documentation/developer-onboarding.md [hunk 18]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+    and organisation
+- 	Select the Appropriate Data Sources by identifying the right datasource to visualise in the dashboard (examples of usual datasources 
+    are prometheus, , databases, APIs, log files)
+-	Create Panels for each metric by adding panel for the metric and choosing the right visualisation (for example graph, gauge, table, 
+    heatmap) and configure the panel settings eg the query, data transformation and diaplay options and add concise title for clarity
+
+###  Changes to dashboards and pull requests
+- Any changes made to the dashboard on the UI will be overwritten in the next deployment unless added to the codebase and a pull request 
+  made to merge it
+- in order to ensure that the new dashboard created is permanent and not deleted by subsequent deployment, add a JSON  file to the
```
Comment (saliceti) [inline documentation/developer-onboarding.md:177]: - In order to

File: documentation/developer-onboarding.md [hunk 19]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+- Import the JSON File by either clicking on "Upload JSON file" and selecting the json file from your computer or pasting the json file 
+  content into the text area provided
+-  Click on the "Import" button to initiate the dashboard import process
+
+### Alertmanager
+The alertmanager urls corresponding to the various environments are
+* Test | https://alertmanager.test.teacherservices.cloud/
+* Production | https://alertmanager.teacherservices.cloud/
+
+Authentication details are usually required but this is stored in the keyvault. Please ask in the teacher-services-infra channel for more details.
```
Comment (saliceti) [inline documentation/developer-onboarding.md:195]: `#teacher-services-infra`

File: documentation/developer-onboarding.md (hunk 12 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:148]: removed

File: documentation/developer-onboarding.md (hunk 13 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:149]: changed

File: documentation/developer-onboarding.md (hunk 14 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:147]: done

File: documentation/developer-onboarding.md (hunk 13 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:149]: changed

File: documentation/developer-onboarding.md (hunk 16 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:169]: changed to only prometheus

File: documentation/developer-onboarding.md (hunk 18 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:177]: corrected

File: documentation/developer-onboarding.md [hunk 20]
```diff
@@ -142,3 +142,54 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+    heatmap) and configure the panel settings eg the query, data transformation and diaplay options and add concise title for clarity
+
+###  Changes to dashboards and pull requests
+- Any changes made to the dashboard on the UI will be overwritten in the next deployment unless added to the codebase and a pull request 
+  made to merge it
+- in order to ensure that the new dashboard created is permanent and not deleted by subsequent deployment, add a JSON  file to the 
+  dashboards directory [here](https://github.com/DFE-Digital/teacher-services-cloud/tree/main/cluster/terraform_kubernetes/config/dashboards) by pasting the content of the json file exported from the dashboard and then make an entry to grafana_dashboards kubernetes_config_map resource in [grafana.tf](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/cluster/terraform_kubernetes/grafana.tf#L152C1-L153C1) file and raise a PR in order to merge the change
+
+
+### Creating a Grafana Dashboard from existing json and updating the code with new dashboard
```
[the remark this answers is not in the document]
Comment (temitope777) [reply documentation/developer-onboarding.md:181]: changed

File: documentation/developer-onboarding.md (hunk 19 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:195]: changed

File: documentation/developer-onboarding.md (hunk 17 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:171]: added

File: documentation/developer-onboarding.md (hunk 11 above)
Comment (saliceti) [reply documentation/developer-onboarding.md:145]: Please add a blank line above

File: documentation/developer-onboarding.md (hunk 12 above)
Comment (saliceti) [reply documentation/developer-onboarding.md:147]: Please add a blank line above

File: documentation/developer-onboarding.md [hunk 21]
```diff
@@ -142,3 +142,53 @@ HTTP requests from the ingress controller, using the filter from [ingress contro
...
+The following steps are required for creating or editing dashboards. Please [click](https://grafana.com/docs/grafana/latest/fundamentals/dashboards-overview/) for more extensive details
+-   Ensure you are logged in as an admin
+-   Identify the purpose of your dashboard. What insights the dashboard will provide and what messages it  conveys
+- 	Plan and Design how the dashboard would look when completed, paying attention to the placement of panels, alignment, spacing, colour 
+    and organisation
+- 	Select the Appropriate Data Sources by identifying the right datasource to visualise in the dashboard (currently prometheus is the only 
+   datasource available to select )
+-  Click on the "Explore" view, select the datasource(prometheus) and then browse and search using the "Metric" dropdown
+-	Create Panels for each metric by adding panel for the metric and choosing the right visualisation (for example graph, gauge, table, 
+    heatmap) and configure the panel settings eg the query, data transformation and diaplay options and add concise title for clarity
```
Comment (saliceti) [inline documentation/developer-onboarding.md:172]: Typo: display

File: documentation/developer-onboarding.md (hunk 11 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:145]: added

File: documentation/developer-onboarding.md (hunk 12 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:147]: added

File: documentation/developer-onboarding.md (hunk 13 above)
Comment (temitope777) [reply documentation/developer-onboarding.md]: added

File: documentation/developer-onboarding.md (hunk 21 above)
Comment (temitope777) [reply documentation/developer-onboarding.md:172]: corrected

**Review by saliceti — APPROVED** [review]

**Pull request merged**